openbas-api/src/main/java/io/openbas/rest/asset_group/AssetGroupCriteriaBuilderService.java

```diff
@@ -39,21 +39,27 @@ public Page<AssetGroupOutput> assetGroupPagination(
   }
 
   public List<AssetGroupOutput> find(Specification<AssetGroup> specification) {
-    CriteriaBuilder cb = this.entityManager.getCriteriaBuilder();
-
-    CriteriaQuery<Tuple> cq = cb.createTupleQuery();
-    Root<AssetGroup> root = cq.from(AssetGroup.class);
-    select(cb, cq, root);
+    try {
+      CriteriaBuilder cb = this.entityManager.getCriteriaBuilder();
+
+      CriteriaQuery<Tuple> cq = cb.createTupleQuery();
+      Root<AssetGroup> root = cq.from(AssetGroup.class);
+      select(cb, cq, root);
+
+      if (specification != null) {
+        Predicate predicate = specification.toPredicate(root, cq, cb);
+        if (predicate != null) {
+          cq.where(predicate);
+        }
+      }
 
-    if (specification != null) {
-      Predicate predicate = specification.toPredicate(root, cq, cb);
-      if (predicate != null) {
-        cq.where(predicate);
+      TypedQuery<Tuple> query = entityManager.createQuery(cq);
+      return execution(query, this.mapper);
+    } finally {
+      if (entityManager != null && entityManager.isOpen()) {
+        entityManager.close();
```

Comment:
The @PersistenceContext handle this
https://stackoverflow.com/questions/17504784/closing-entitymanager-by-spring